src/compiler/watch.ts

```diff
@@ -138,109 +171,290 @@ namespace ts {
-            let sourceMaps: SourceMapData[];
-            let emitSkipped: boolean;
+        function writeFile(fileName: string, text: string, writeByteOrderMark: boolean, onError: (message: string) => void) {
+            try {
+                performance.mark("beforeIOWrite");
+                ensureDirectoriesExist(getDirectoryPath(normalizePath(fileName)));
 
-            const result = builder.emitChangedFiles(program, writeFile);
-            if (result.length === 0) {
-                emitSkipped = true;
-            }
-            else {
-                for (const emitOutput of result) {
-                    if (emitOutput.emitSkipped) {
-                        emitSkipped = true;
-                    }
-                    addRange(diagnostics, emitOutput.diagnostics);
-                    sourceMaps = concatenate(sourceMaps, emitOutput.sourceMaps);
-                }
-            }
+                host.writeFile(fileName, text, writeByteOrderMark);
 
-            if (reportSemanticDiagnostics) {
-                addRange(diagnostics, builder.getSemanticDiagnostics(program));
+                performance.mark("afterIOWrite");
+                performance.measure("I/O Write", "beforeIOWrite", "afterIOWrite");
             }
-            return handleEmitOutputAndReportErrors(host, program, emittedFiles, emitSkipped,
-                diagnostics, reportDiagnostic);
-
-            function ensureDirectoriesExist(directoryPath: string) {
-                if (directoryPath.length > getRootLength(directoryPath) && !host.directoryExists(directoryPath)) {
-                    const parentDirectory = getDirectoryPath(directoryPath);
-                    ensureDirectoriesExist(parentDirectory);
-                    host.createDirectory(directoryPath);
+            catch (e) {
+                if (onError) {
+                    onError(e.message);
                 }
             }
+        }
+    }
 
-            function writeFile(fileName: string, text: string, writeByteOrderMark: boolean, onError: (message: string) => void) {
-                try {
-                    performance.mark("beforeIOWrite");
-                    ensureDirectoriesExist(getDirectoryPath(normalizePath(fileName)));
+    /**
+     * Report error and exit
+     */
+    function reportUnrecoverableDiagnostic(system: System, reportDiagnostic: DiagnosticReporter, diagnostic: Diagnostic) {
+        reportDiagnostic(diagnostic);
+        system.exit(ExitStatus.DiagnosticsPresent_OutputsSkipped);
+    }
 
-                    host.writeFile(fileName, text, writeByteOrderMark);
+    /**
+     * Creates the watch compiler host from system for config file in watch mode
+     */
+    export function createWatchCompilerHostOfConfigFile(configFileName: string, optionsToExtend: CompilerOptions | undefined, system: System, reportDiagnostic: DiagnosticReporter | undefined, reportWatchStatus: WatchStatusReporter | undefined): WatchCompilerHostOfConfigFile {

```

Comment:
consider combining these two APIs into `createWatchCompilerHost` (jsut like we have`createCompilerHost`) with two overloads instead of two separate functions.

Reply:
done